php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -50,6 +58,104 @@ public ExpressionTree EXPRESSION() {
+          b.token(PHPLexicalGrammar.TRAIT_CONSTANT)))));
+  }
+
+  public LiteralTree NUMERIC_LITERAL() {
+    return b.<LiteralTree>nonterminal()
+      .is(f.numericLiteral(b.token(PHPLexicalGrammar.NUMERIC_LITERAL)));
+  }
+
+  public ExpandableStringLiteralTree EXPANDABLE_STRING_LITERAL() {
+    return b.<ExpandableStringLiteralTree>nonterminal(PHPLexicalGrammar.ENCAPS_STRING_LITERAL)
+      .is(f.expandableStringLiteral(
+        b.token(PHPLexicalGrammar.SPACING),
+        b.token(PHPLexicalGrammar.DOUBLE_QUOTE),
+        b.oneOrMore(
+          b.firstOf(
+            ENCAPSULATED_STRING_VARIABLE(),
+            EXPANDABLE_STRING_CHARACTERS()
+          )
+        ),
+        b.token(PHPLexicalGrammar.DOUBLE_QUOTE)
+      ));
+  }
+
+  public ExpressionTree ENCAPSULATED_STRING_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.ENCAPS_VAR)
+      .is(b.firstOf(
+          ENCAPSULATED_SEMI_COMPLEX_VARIABLE(),
+          ENCAPSULATED_SIMPLE_VARIABLE(),
+          ENCAPSULATED_COMPLEX_VARIABLE()
+        )
+      );
+  }
+
+  public ExpressionTree ENCAPSULATED_COMPLEX_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.COMPLEX_ENCAPS_VARIABLE)
+      .is(f.encapsulatedComplexVariable(
+        b.token(PHPPunctuator.LCURLYBRACE),
+        b.token(PHPLexicalGrammar.NEXT_IS_DOLLAR),
+        EXPRESSION(),
+        b.token(PHPPunctuator.RCURLYBRACE)
+      ));
+  }
+
+  public ExpressionTree ENCAPSULATED_SEMI_COMPLEX_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.SEMI_COMPLEX_ENCAPS_VARIABLE)
+      .is(f.encapsulatedSemiComplexVariable(
+        b.token(PHPPunctuator.DOLLAR_LCURLY),
+        b.firstOf(
+          EXPRESSION(),
+          f.expressionRecovery(b.token(PHPLexicalGrammar.SEMI_COMPLEX_RECOVERY_EXPRESSION))),
+        b.token(PHPPunctuator.RCURLYBRACE)));
+  }
+
+  public ExpressionTree ENCAPSULATED_SIMPLE_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.SIMPLE_ENCAPS_VARIABLE)
+    .is(
+      f.encapsulatedSimpleVar(ENCAPSULATED_VARIABLE_IDENTIFIER(),
+        b.optional(b.firstOf(
+          f.expandableArrayAccess(
+            b.token(PHPPunctuator.LBRACKET),
+            b.firstOf(IDENTIFIER(), NUMERIC_LITERAL(), ENCAPSULATED_VARIABLE_IDENTIFIER()),
+            b.token(PHPPunctuator.RBRACKET)),
+          f.expandableObjectMemberAccess(b.token(PHPPunctuator.ARROW), IDENTIFIER()))))
+    );
+  }
+
+  public IdentifierTree IDENTIFIER() {
+    return b.<IdentifierTree>nonterminal()
+      .is(f.identifier(b.token(PHPLexicalGrammar.IDENTIFIER)));
+  }
+
+  public VariableIdentifierTree ENCAPSULATED_VARIABLE_IDENTIFIER() {
+    return b.<VariableIdentifierTree>nonterminal(PHPLexicalGrammar.ENCAPS_VAR_IDENTIFIER)
+      .is(f.encapsulatedVariableIdentifier(b.token(PHPLexicalGrammar.WHITESPACES), b.token(PHPLexicalGrammar.VARIABLE_IDENTIFIER)));
+  }
+
+  public ExpressionTree EXPANDABLE_STRING_CHARACTERS() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)